torchtune/data/_messages.py

```diff
@@ -129,13 +129,19 @@ class InputOutputToMessages(Transform):
         column_map (Optional[Dict[str, str]]): a mapping to change the expected "input"
             and "output" column names to the actual column names in the dataset. Default is None,
             keeping the default "input" and "output" column names.
+        system_prompt (Optional[str]): if specified, prepend a system message. This can
+            serve as instructions to guide the model response. Default is None.
     """
 
     def __init__(
-        self, train_on_input: bool = False, column_map: Optional[Dict[str, str]] = None
+        self,
+        train_on_input: bool = False,
+        column_map: Optional[Dict[str, str]] = None,
+        system_prompt: Optional[str] = None,
```

Comment:
Need some way to convey that this will either add or override the given system prompt:

```new_system_prompt```?